packages/logger/src/Logger.ts

```diff
@@ -934,7 +958,33 @@ class Logger extends Utility implements LoggerInterface {
           this.createAndPopulateLogItem(logLevel, input, extraInput)
         );
       } else {
-        this.#buffer.push([logLevel, [logLevel, input, extraInput]]);
+        this.#initBuffer.push([logLevel, [logLevel, input, extraInput]]);
+      }
+      return;
+    }
+
+    const trace_id = this.envVarsService.getXrayTraceId();
+    if (trace_id !== undefined && this.shouldBufferLog(trace_id, logLevel)) {
```

Comment:
I needed to check `trace_id` was not undefined here to satisfy the type checks. It was originally just being checked in `shouldBufferLog`. Ideally the check would be in just `shouldBufferLog`. There's probably a better way to do this - it might be a typescript knowledge issue

Reply:
I've left it as is